utils/src/structs.rs

```diff
@@ -30,25 +30,63 @@ pub struct Manifest {
     /// Human-readable description.
     description: String,
     /// Required extensions.
-    required_extensions: Vec<String>, // TODO: this is a placeholder
+    required_extensions: Vec<String>,
+    // TODO: this is a placeholder and requires more thought; the WASM binary itself contains the
+    // info we need to enumerate the required extensions it is looking for. However, for job
+    // routing, it would be great for this information to be available without having the binary
+    // on-hand locally. The right answer here is probably to make this field be optional in manifest
+    // files, and to derive the value automatically at binary/manifest storage time.
+    /// Required permissions; it is up to the agent to ensure that the submitter of this job is
+    /// actually authorized to run a job with said permissions.
+    required_permissions: Vec<Permission>,
 }
 
 impl Manifest {
+    pub fn new(path: &PathBuf) -> Manifest {
+        Manifest {
+            name: path.file_stem().unwrap().to_string_lossy().to_string(),
+            namespace: String::from(""),
+            binary: path.to_owned(),
+            version: String::from("0.0.0"),
+            description: String::from(""),
+            required_extensions: vec![],
+            required_permissions: vec![],
+        }
+    }
+
     pub fn from_string(input: &str) -> Result<Self, ServalError> {
         let manifest: Manifest = toml::from_str(input)?;
+        if manifest.binary.is_relative() {
+            return Err(ServalError::RelativeBinaryPathInManifestError);
```

Comment:
The CLI currently assumes that a Wasm path is _always_ relative (and in fact, a file in the same directory as the manifest file itself). [see here](https://github.com/serval/serval-mesh/blob/main/cli/src/main.rs#L94)
Absolute paths don't really make sense in a manifest considering that both the manifest and the binary end up in the mesh storage anyways, rendering the path obsolete.
Actually... shouldn't there simply be a convention for where the Wasm binary should be relative to the manifest? I feel like the way the CLI handles it is too much of an explicit manipulation as-is, but it shouldn't really be absolute, either.

Reply:
ping: @ceejbot probably has opinions here

Reply:
Ah, crap. A file obviously can't be relative to a string, I get why this must throw an error.
Still... Do we then ultimately store the manifest in the mesh with an absolute path that is invalid on all systems except the originating one? 🤔

Reply:
The manifests we store in the mesh shouldn't have binary paths at all; they should have content-addressable storage hashes. Binary paths only make sense when running locally (e.g. when you're in the act of storing a job, or when you're using the test-runner).